app/workers/functions.php

```diff
@@ -412,48 +371,61 @@ public function execute(string $trigger, string $projectId, string $executionId,
             $cpus = App::getEnv('_APP_FUNCTIONS_CPUS', '');
             $memory = App::getEnv('_APP_FUNCTIONS_MEMORY', '');
             $swap = App::getEnv('_APP_FUNCTIONS_MEMORY_SWAP', '');
-            $exitCode = Console::execute("docker run ".
-                " -d".
-                " --entrypoint=\"\"".
-                (empty($cpus) ? "" : (" --cpus=".$cpus)).
-                (empty($memory) ? "" : (" --memory=".$memory."m")).
-                (empty($swap) ? "" : (" --memory-swap=".$swap."m")).
-                " --name={$container}".
-                " --label appwrite-type=function".
-                " --label appwrite-created={$executionTime}".
-                " --volume {$tagPathTargetDir}:/tmp:rw".
-                " --workdir /usr/local/src".
-                " ".\implode(" ", $vars).
-                " {$runtime['image']}".
-                " tail -f /dev/null"
-            , '', $stdout, $stderr, 30);
-
-            if($exitCode !== 0) {
-                throw new Exception('Failed to create function environment: '.$stderr);
-            }
 
-            $exitCodeUntar = Console::execute("docker exec ".
-                $container.
-                " sh -c 'mv /tmp/code.tar.gz /usr/local/src/code.tar.gz && tar -zxf /usr/local/src/code.tar.gz --strip 1 && rm /usr/local/src/code.tar.gz'"
-                , '', $stdout, $stderr, 60);
+            $orchestration->setCpus($cpus);
+            $orchestration->setMemory($memory);
+            $orchestration->setSwap($swap);
+
+            \array_walk($vars, function (string &$value) {
+                $value = strval($value);
+            });
+
+            $id = $orchestration->run(
+                $runtime['image'],
+                $container,
+                '',
```

Comment:
I know this might become a little weird, but would improve readabilty by 200%.

Let's used named arguments for method such as this - where there are many arguments that doesn't really tell you what they do.

ref: https://stitcher.io/blog/php-8-named-arguments